### PR TITLE
Update events_cache.go

### PR DESCRIPTION
**Pull request by HeGaoYuan**
With more accurate comment

#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
I think the origin comment is not consistent with the code.
The comment says during 10 minutes, if there are more than 10 events, then we aggregate the event.
If the event arises every 9 minutes, so during 10 minutes there are only 1 event, this is obviously not conform to the comment, so we should not aggregate the event, but the code logic will aggregate the event

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/record/events_cache.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ const (
 	maxLruCacheEntries = 4096
 
 	// if we see the same event that varies only by message
-	// more than 10 times in a 10 minute period, aggregate the event
+	// more than 10 times, which the max interval of two events are 10 minutes,
+	// then we aggregate the event
```
Comment (heqg) [inline staging/src/k8s.io/client-go/tools/record/events_cache.go:41]: /lgtm

File: staging/src/k8s.io/client-go/tools/record/events_cache.go [hunk 2]
```diff
@@ -37,7 +37,8 @@ const (
 	maxLruCacheEntries = 4096
 
 	// if we see the same event that varies only by message
-	// more than 10 times in a 10 minute period, aggregate the event
+	// more than 10 times, which the max interval of two events are 10 minutes,
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/record/events_cache.go:40]: Do you mean "...more than 10 times, and the maximum interval between any two adjacent events is below 10 minutes,..."?

Then this [comment](https://github.com/kubernetes/kubernetes/blob/1945829906546caf867992669a0bfa588edf8be6/staging/src/k8s.io/client-go/tools/record/events_cache.go#L430-L434) is also inaccurate. It mentions the 10 minute "rolling interval".

Comment (HeGaoYuan) [reply staging/src/k8s.io/client-go/tools/record/events_cache.go]: Yes, that's what I mean.
I also fixed the another comment you mentioned.

**Comment by HeGaoYuan** [comment]
@caesarxuchao
Please have a review again.

**Comment by caesarxuchao** [comment]
Can you add a test case to verify the behavior? e.g., verify that 10 events that are `defaultAggregateIntervalInSeconds-1` seconds apart are aggregated.

I'm considering whether it's worth fixing the code to match the original description of the comment. Perhaps that's unnecessary, as the event cache has behaved like this for years and no one has complained.

**Comment by HeGaoYuan** [comment]
> Can you add a test case to verify the behavior? e.g., verify that 10 events that are `defaultAggregateIntervalInSeconds-1` seconds apart are aggregated.
> 
> I'm considering whether it's worth fixing the code to match the original description of the comment. Perhaps that's unnecessary, as the event cache has behaved like this for years and no one has complained.

@caesarxuchao 
I've added some test case.
And I found the test code have some problems about the clock. If we use the SimpleIntervalClock and pass it to NewEventCorrelator function, each time the inside of correlator call the clock.Now() will steps the clock forward the specified duration, it is not expected. I changed the SimpleIntervalClock to FakePassiveClock, and manually control the time in the for loop.

**Comment by HeGaoYuan** [comment]
@caesarxuchao 
I'm worried that you forgot so I remind you

**Comment by serathius** [comment]
cc @olagacek

File: staging/src/k8s.io/client-go/tools/record/events_cache_test.go [hunk 3]
```diff
@@ -228,14 +240,26 @@ func TestEventCorrelator(t *testing.T) {
 			previousEvents:  makeSimilarEvents(defaultAggregateMaxEvents-1, similarEvent, similarEvent.Message),
 			newEvent:        similarEvent,
 			expectedEvent:   setCount(similarEvent, 1),
-			intervalSeconds: defaultAggregateIntervalInSeconds,
+			intervalSeconds: defaultAggregateIntervalInSeconds + 1,
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/record/events_cache_test.go:243]: Why do we change this line?

File: staging/src/k8s.io/client-go/tools/record/events_cache_test.go [hunk 4]
```diff
@@ -228,14 +240,26 @@ func TestEventCorrelator(t *testing.T) {
...
+			previousEvents:  makeSimilarEvents(defaultAggregateMaxEvents-1, similarEvent, similarEvent.Message),
+			newEvent:        similarEvent,
+			expectedEvent:   setCount(aggregateEvent, 1),
+			intervalSeconds: defaultAggregateIntervalInSeconds - 1,
+		},
+		"similar-events-whose-interval-is-defaultAggregateIntervalInSecondsMinusOneSecond-should-count-the-aggregate": {
+			previousEvents:  makeSimilarEvents(defaultAggregateMaxEvents, similarEvent, similarEvent.Message),
+			newEvent:        similarEvent,
+			expectedEvent:   setCount(aggregateEvent, 2),
+			intervalSeconds: defaultAggregateIntervalInSeconds - 1,
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/record/events_cache_test.go:255]: Can we just reuse the "similar-events-should-aggregate-event" and "similar-events-many-times-should-count-the-aggregate", just change their intervalSeconds to `defaultAggregateIntervalInSeconds-1`.

File: staging/src/k8s.io/client-go/tools/record/events_cache_test.go [hunk 5]
```diff
@@ -228,14 +240,26 @@ func TestEventCorrelator(t *testing.T) {
...
 		},
 	}
 
 	for testScenario, testInput := range scenario {
 		eventInterval := time.Duration(testInput.intervalSeconds) * time.Second
-		clock := testclocks.SimpleIntervalClock{Time: time.Now(), Duration: eventInterval}
-		correlator := NewEventCorrelator(&clock)
+		clock := testclocks.NewFakePassiveClock(time.Now())
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/record/events_cache_test.go:261]: Why do we change this?

File: staging/src/k8s.io/client-go/tools/record/events_cache_test.go [hunk 6]
```diff
@@ -338,6 +363,7 @@ func TestEventSpamFilter(t *testing.T) {
 			correlator.UpdateState(result.Event)
 		}
 
+		clock.SetTime(clock.Now().Add(eventInterval))
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/record/events_cache_test.go:366]: Why do we change this?

**Pull request closed without merging**